src/components/Button/button.test.tsx

```diff
@@ -163,22 +163,42 @@ describe('<Button />', () => {
     });
   });
 
-  describe('when clicked', () => {
+  describe('with mouse events', () => {
     let clicked: boolean;
-    function clickMe() {
+    let hoveredIn: boolean;
+    let hoveredOut: boolean;
+    function onClick() {
       clicked = true;
     }
+    function onHoverIn() {
+      hoveredIn = true;
+    }
+    function onHoverOut() {
+      hoveredOut = true;
+    }
     beforeEach(() => {
       clicked = false;  
+      hoveredIn = false;  
+      hoveredOut = false;
       component = shallow(
-        <Button text={sampleText} onClick={clickMe} />,
+        <Button text={sampleText} onClick={onClick} onHoverIn={onHoverIn} onHoverOut={onHoverOut} />,
```

Comment:
Consider batching up the props in an object and then using spread to make it more concise as props are added:

```javascript
<Button {...props} />
```